orc/src/main/java/com/netflix/iceberg/orc/TypeConversion.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2018 Hortonworks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.iceberg.orc;
+
+import com.netflix.iceberg.Schema;
+import com.netflix.iceberg.types.Type;
+import com.netflix.iceberg.types.Types;
+import org.apache.orc.TypeDescription;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TypeConversion {
+
+  /**
+   * Convert a given Iceberg schema to ORC.
+   * @param schema the Iceberg schema to convert
+   * @param columnIds an output with the column ids
+   * @return the ORC schema
+   */
+  public static TypeDescription toOrc(Schema schema,
+                                      List<Integer> columnIds) {
+    columnIds.add(0);
+    return toOrc(schema.asStruct(), columnIds);
+  }
+
+  static TypeDescription toOrc(Type type, List<Integer> columnIds) {
```

Comment:
You might consider using a [`SchemaVisitor`](https://github.com/Netflix/iceberg/blob/master/api/src/main/java/com/netflix/iceberg/types/TypeUtil.java#L160-L186) for conversion to separate the conversion logic from type traversal. Here's an example for Avro: https://github.com/Netflix/iceberg/blob/master/avro/src/main/java/com/netflix/iceberg/avro/TypeToSchema.java

Reply:
I looked at the SchemaVisitor. I found it far more difficult to read than having a single function that did the recursion.

Reply:
Sounds reasonable. I think it's important for more complicated transformations, but this does look straight-forward.